contrib/views/form_tabs/curriculum.py

```diff
@@ -23,194 +23,331 @@
-                    uuid=proposition_uuid,
-                )
-            except ApiException:
-                messages.error(self.request, _("An error has happened when deleting the experience."))
-                return self.re_render_after_invalid_form(self.get_context_data())
-
-        return self.redirect_after_valid_form()
-
-    @classmethod
-    def prepare_data(cls, data):
-        # Remove redundant data
-        redundant_fields = [
-            'other_program',
-            'program_not_found',
-            'institute_not_found',
-            'institute_city_be',
-            'activity_institute_city',
-            'activity_institute_name',
+    def prepare_data(self, base_form, year_formset):
+        data = base_form.cleaned_data
+        data['educationalexperienceyear_set'] = [
+            year_data for year_data in year_formset.cleaned_data if year_data.pop('is_enrolled')
         ]
-        for f in redundant_fields:
-            data.pop(f, None)
 
-        # Format some fields values
-        data['academic_year'] = int(data['academic_year'])
-        if data.get('dissertation_score'):
-            data['dissertation_score'] = str(data['dissertation_score'])
+        data.pop('other_institute')
+        data.pop('other_program')
 
         return data
 
-    def redirect_after_valid_form(self):
-        messages.info(self.request, _('The curriculum has correctly been updated.'))
-        return HttpResponseRedirect(self.get_success_url())
+    def check_forms(self, base_form, year_formset):
+        # Individual form check
+        base_form.is_valid()
+        year_formset.is_valid()
+
+        at_least_one_successful_year = False
+        at_least_one_enrolled_year = False
+
+        country = base_form.cleaned_data.get('country')
+        be_country = country == BE_ISO_CODE
+        linguistic_regime = base_form.cleaned_data.get('linguistic_regime')
+        credits_are_required = base_form.cleaned_data.get('evaluation_type') in EvaluationSystemsWithCredits
+        transcript_is_required = base_form.cleaned_data.get('transcript_type') == TranscriptType.ONE_A_YEAR.name
+        transcript_translation_is_required = (
+            transcript_is_required
+            and country
+            and not be_country
+            and linguistic_regime
+            and linguistic_regime not in LINGUISTIC_REGIMES_WITHOUT_TRANSLATION
+        )
+
+        # Cross-form check
+        for form in year_formset:
+            cleaned_data = form.cleaned_data
+            if cleaned_data.get('is_enrolled'):
+                at_least_one_enrolled_year = True
+
+                if cleaned_data.get('result') in SuccessfulResults:
+                    at_least_one_successful_year = True
+
+                # Credit fields
+                if (
+                    credits_are_required
+                    if not be_country
+                    else cleaned_data.get('academic_year') >= FIRST_YEAR_WITH_ECTS_BE
+                ):
```

Comment:
J'inverserais la condition pour éviter le not
`                if cleaned_data.get('academic_year') >= FIRST_YEAR_WITH_ECTS_BE if be_country else credits_are_required:
`